File: src/main/java/org/dementhium/event/Tickable.java
package org.dementhium.event;
/**
 * 
 * @author 'Mystic Flow
 */
public abstract class Tickable {
	
	private int cycles;
	
	private int attempts;
	private boolean running = true;
	
	public Tickable(int cycles) {
		this.cycles = cycles;
		this.attempts = 0;
	}
	
	public void run() {
		attempts++;
		if(attempts >= cycles) {
			attempts = 0;
			execute();
		}
	}
	
	public boolean isRunning() {
		return running;
	}
	
	public void setTicks(int cycles) {
		this.cycles = cycles;
	}
	
	public void stop() {
		running = false;
	}
	
	public abstract void execute();
	
}

File: src/main/java/org/dementhium/event/Tickable.kt
package org.dementhium.event

abstract class Tickable(var cycles: Int) {

    var attempts: Int = 0

    var running = true

    fun isRunning(): Boolean = running

    open fun stop() { running = false }

    fun run() {
        if (attempts++ >= cycles) {
            attempts = 0
            execute()
        }
    }

    abstract fun execute()
}
